src/ext/serial.cpp

```diff
@@ -4,17 +4,45 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#ifdef __MBED__
+#ifndef UNIT_TEST
 
-#include "mbed.h"
 #include "config.h"
 #include "thingset.h"
 #include "ext.h"
 
+#ifdef __MBED__
+
+#include "mbed.h"
+
+#elif defined(__ZEPHYR__)
+
+#include <zephyr.h>
+#include <sys/printk.h>
+#include <console/console.h>
+#include <string.h>
+#include <stdio.h>
+#include <drivers/uart.h>
+
+#define UART_DEVICE_NAME CONFIG_UART_CONSOLE_ON_DEV_NAME
+struct device *uart_dev = device_get_binding(UART_DEVICE_NAME);
```

Comment:
There is now an uart-dbg alias defined in the board DTS, which you can access with `DT_ALIAS_UART_DBG_LABEL` (instead of `CONFIG_UART_CONSOLE_ON_DEV_NAME`). This allows to put ThingSet to a different UART interface than the console.